src/Qir/Common/Include/SimulatorStub.hpp

```diff
@@ -1,139 +1,139 @@
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledX(long numControls, Qubit controls[], Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledY(long numControls, Qubit controls[], Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledZ(long numControls, Qubit controls[], Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledH(long numControls, Qubit controls[], Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledS(long numControls, Qubit controls[], Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledT(long numControls, Qubit controls[], Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledR(long numControls, Qubit controls[], PauliId axis, Qubit target, double theta) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledExp(
-            long numControls,
-            Qubit controls[],
-            long numTargets,
-            PauliId paulis[],
-            Qubit targets[],
-            double theta) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void AdjointS(Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void AdjointT(Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledAdjointS(long numControls, Qubit controls[], Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ControlledAdjointT(long numControls, Qubit controls[], Qubit target) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        Result Measure(long numBases, PauliId bases[], long numTargets, Qubit targets[]) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        void ReleaseResult(Result result) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        bool AreEqualResults(Result r1, Result r2) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        ResultValue GetResultValue(Result result) override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        Result UseZero() override
-        {
-            throw std::logic_error("not_implemented");
-        }
-        Result UseOne() override
-        {
-            throw std::logic_error("not_implemented");
-        }
-    };
-
-} // namespace Quantum
+#pragma once
```

Comment:
It seems like this file and a few others did not change the contents but have now different line endings. This might have happened because the file was moved "outside" of git and even though it was detected as a rename, line endings were not preserved.

Please consider using the `git mv` command which does preserve line endings.

Reply:
I double checked, and unfortunately this is the right thing in this case. This repository was missing a `.gitattributes` file that would ensure that line endings get converted to Unix style when checked in to avoid this kind of issue. I added that file last month (see #545), but these files were created before then. This means that our repo has a mix of Unix line endings and Windows line endings in it, and someone cloning the repo on Linux would see every file with `CRLF` line endings as modified because git will convert them `LF`. Now that we have the gitattributes, future files will correctly be merged with Unix line endings but checked out locally with `CRLF` line endings if the local environment is Windows. Now that this is set, the first time we move a file git will see the line endings change, which is unfortunate but sets us up for better cross-platform development in the future.

See [Configuring Git to handle line endings](https://docs.github.com/en/github/getting-started-with-github/configuring-git-to-handle-line-endings) for more details on the guidance here. We should be making this change across all our repos to better enable cross-platform collaboration.